### DISCUSSION TITLE
How to Configure Bob's Window to Float Instead of Opening in a New Workspace

### DISCUSSION THREAD

**Discussion by yaotutu** [nikitabobko/AeroSpace#1005] [potential-bugs]
### Description
i am trying to configuer the window management for an application named "Bob" using Aerospace. My foal is to make the Bob applicaion window float whenecer it appears,instead of having it open in a new workspace.

Here is the current configuarion i am useing :
``` toml
[[on-window-detected]]
if.app-id = 'com.hezongyidev.Bob'
check-further-callbacks = true
run = 'layout floating'

```
### Problem:
Despite this configuration, Bob’s window still opens in a new workspace rather than floating. I’ve confirmed the app ID is correct using the following command:

```shell

mdls -name kMDItemCFBundleIdentifier -r /Applications/bob.app
```
This returns: com.hezongyidev.Bob

### Hypothesis:
I suspect the issue may be related to how Bob operates. The application primarily uses floating windows to display information, and it’s possible that these floating windows are invoked through a system-level API that doesn’t directly associate them with Bob’s main application process. This might explain why Aerospace isn’t handling them as expected. If the floating window is technically not part of the bob application (from the perspective of the window manager), it might not be applying the expected layout.

<img width="1630" alt="image" src="https://github.com/user-attachments/assets/e5e8cd57-bf2e-4880-bda8-544f666dabe9">

**Comment by tobiasgiese** [comment]
Have you tried to get the identifier via `aerospace list-apps`?

**Comment by yaotutu** [comment]
@tobiasgiese 
> Have you tried to get the identifier via `aerospace list-apps`?

I have tried using the aerospace list-apps command, but unfortunately, it does not list the Bob application. It seems that Bob’s window or application might not be recognized or registered by Aerospace in the way other applications are.

**Comment by nikitabobko** [comment]
Please attach `aerospace debug-windows` output

**Comment by yaotutu** [comment]
Bob’s app is a floating window. When the focus disappears, the window also disappears. Therefore, this command does not provide any useful information.

**Comment by martinengelke** [comment]
I can underline the issue with an app called SideNote, which is normally following spaces on macOS to appear right at the side of the screen like the Notification Center. 
Now when I invoke it, it just prompts me to a certain Workspace where it then opens. 
The fundamental idea of SideNotes of following my spaces is unfortunately gone :( 

Maybe there is some way to unleash an app from a workspace and make it visible for all workspaces? 

Edit: could work with:

On activate:
Bring to current workspace

**Comment by d3vw** [comment]
> ### Description
> 
> i am trying to configuer the window management for an application named "Bob" using Aerospace. My foal is to make the Bob applicaion window float whenecer it appears,instead of having it open in a new workspace.
> 
> Here is the current configuarion i am useing :
> 
> ```toml
> [[on-window-detected]]
> if.app-id = 'com.hezongyidev.Bob'
> check-further-callbacks = true
> run = 'layout floating'
> ```
> 
> ### Problem:
> 
> Despite this configuration, Bob’s window still opens in a new workspace rather than floating. I’ve confirmed the app ID is correct using the following command:
> 
> ```shell
> mdls -name kMDItemCFBundleIdentifier -r /Applications/bob.app
> ```
> 
> This returns: com.hezongyidev.Bob
> ### Hypothesis:
> 
> I suspect the issue may be related to how Bob operates. The application primarily uses floating windows to display information, and it’s possible that these floating windows are invoked through a system-level API that doesn’t directly associate them with Bob’s main application process. This might explain why Aerospace isn’t handling them as expected. If the floating window is technically not part of the bob application (from the perspective of the window manager), it might not be applying the expected layout.
> <img alt="image" width="1630" src="https://private-user-images.githubusercontent.com/21394924/357692967-e5e8cd57-bf2e-4880-bda8-544f666dabe9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU5MDgxODMsIm5iZiI6MTc0NTkwNzg4MywicGF0aCI6Ii8yMTM5NDkyNC8zNTc2OTI5NjctZTVlOGNkNTctYmYyZS00ODgwLWJkYTgtNTQ0ZjY2NmRhYmU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI5VDA2MjQ0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYxNTllNDAyZTY1NzkxMjBjNGQ4NmEzMDhkOTdiNGI1NzQ4ZjhlYjliNDQ4MDQ3N2E0YjQ2NDlhNDRjODBmMDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7wvvz9pmjfoalucA-kMxMCK0dCgjSv-UaUURmIVUGqM">

Hi, I found the reason why the Bob panel kept showing up in the next workspace: it's because of a shortcut conflict! The shortcut `Option + A/S/D` is used by Bob, but it's also used by AeroSpace.

The solution is to remove those shortcuts from the `.aerospace.toml` config file.

Here's my current config — I kept only `Option + 0~9`.

<img width="944" alt="image" src="https://github.com/user-attachments/assets/e3bb8959-1898-4b57-bace-e2289c722bef" />
